server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -287,21 +343,42 @@ public SourceFieldMapper build() {
 
     }
 
+    private static SourceFieldMapper resolveSourceMode(final IndexMode indexMode, final Mode sourceMode, boolean enableRecoverySource) {
+        if (indexMode == IndexMode.STANDARD) {
+            return enableRecoverySource ? DEFAULT : DEFAULT_NO_RECOVERY_SOURCE;
+        }
+        final SourceFieldMapper syntheticWithoutRecoverySource = indexMode == IndexMode.TIME_SERIES
+            ? TSDB_DEFAULT_NO_RECOVERY_SOURCE
+            : LOGSDB_DEFAULT_NO_RECOVERY_SOURCE;
+        final SourceFieldMapper syntheticWithRecoverySource = indexMode == IndexMode.TIME_SERIES ? TSDB_DEFAULT : LOGSDB_DEFAULT;
+        final SourceFieldMapper storedWithoutRecoverySource = indexMode == IndexMode.TIME_SERIES
+            ? TSDB_DEFAULT_NO_RECOVERY_SOURCE_STORED
+            : LOGSDB_DEFAULT_NO_RECOVERY_SOURCE_STORED;
+        final SourceFieldMapper storedWithRecoverySource = indexMode == IndexMode.TIME_SERIES ? TSDB_DEFAULT_STORED : LOGSDB_DEFAULT_STORED;
+
+        switch (sourceMode) {
+            case SYNTHETIC:
+                return enableRecoverySource ? syntheticWithRecoverySource : syntheticWithoutRecoverySource;
+            case STORED:
+                return enableRecoverySource ? storedWithRecoverySource : storedWithoutRecoverySource;
+            case DISABLED:
+                throw new IllegalArgumentException("_source cannot be disabled in index using [" + indexMode + "] index mode");
```

Comment:
This is where two of the error messages have changed slightly and instead of using `can not` (two words) we use `cannot` (single word).